src/sage/rings/finite_rings/integer_mod.pyx

```diff
@@ -1419,7 +1420,7 @@ cdef class IntegerMod_abstract(FiniteRingElement):
             ....:         if (y^41).nth_root(41*r)**(41*r) != y^41: raise RuntimeError
             ....:         if (y^307).nth_root(307*r)**(307*r) != y^307: raise RuntimeError
 
-            sage: for t in range(200):
+            sage: for t in range(200):                                                  # optional - sage.libs.pari
             ....:     n = randint(1,2^63)
             ....:     K = Integers(n)
             ....:     b = K.random_element()
```

Comment:
Is this right? Why `pari`?

Reply:
PARI is called by `nth_root` via `sage.rings.factorint`

Reply:
There is no `nth_root` here..

`sage.rings.factorint` is used in `K = Integers(n)` or `K.random_element()`?

Reply:
The body of this `for` loop continues a few more lines...

Reply:
I see. Sorry.